apollo-backend/src/main/java/io/logz/apollo/kubernetes/KubernetesMonitor.java

```diff
@@ -111,16 +112,12 @@ public void monitor() {
                         returnedDeployment = kubernetesHandler.cancelDeployment(deployment);
                         break;
                     default:
-                        if (deployment.getGroupName() == null) {
-                            returnedDeployment = kubernetesHandler.monitorDeployment(deployment);
+                        if (deployment.getGroupName() != null
+                                && groupDao.getGroupByName(deployment.getGroupName()) != null
```

Comment:
This check can only fail if someone is manually messing with the DB, and meant to prevent NPE in that case. It is probably redundant.

Reply:
Checking for NPE is never redundant, but there is just no point in trying to monitor that deployment. Just throw something

Reply:
The reason I think it's redundant is that it can only happen *if someone is manually messing with the DB* - in case you deploy on a group and delete it right after that, and then the deployment is trying to be monitored.
It's super edgy and unreasonable.